pkg/infrabin/http_test.go

```diff
@@ -265,6 +267,44 @@ func TestProxyHandler(t *testing.T) {
 	}
 }
 
+func TestProxyHandlerRegexpDenyURL(t *testing.T) {
+	// Set the Proxy to true for testing
+	viper.Set("proxyEndpoint", true)
+	viper.Set("proxyAllowRegexp", "fakeurl")
+
+	response, err := json.Marshal(map[string]string{"foo": "bar"})
+	if err != nil {
+		t.Fatalf("Failed to marshal fake response: %v", err)
+	}
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		if _, err := w.Write(response); err != nil {
+			t.Fatalf("Failed to write fake response body: %v", err)
+		}
+	}))
+	defer mockServer.Close()
```

Comment:
I don't think we actually need any of this as the `/proxy` endpoint should return before attempting to call this.